### PR TITLE
Improve annotations for XLinq methods taking params object[]

### LINKED ISSUE #43416: Nullability annotations in XElement constructor
After downloading the latest .Net 5 SDK (Preview 2) I'm starting to update my code to it. 

I'm using nullable reference types. 

I think the nullability annotations could be improved in https://github.com/dotnet/runtime/blob/035b729d829368c2790d825bd02db14f0c0fd2ea/src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XElement.cs#L95

```C#
public XElement(XName name, params object[] content) : this(name, (object)content) { } //Currently
public XElement(XName name, params object?[] content) : this(name, (object)content) { } //Fixed
```

### PR DESCRIPTION
**Pull request by krwq**
Fixes: https://github.com/dotnet/runtime/issues/43416

plus similar APIs across XLinq

cc: @jeffhandley for bar-check

### CODE REVIEW

**Comment by stephentoub** [comment]
I don't see the reference assemblies being updated...?

**Comment by krwq** [comment]
@stephentoub, right, forgot about the ref as it's in different folder

File: src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNode.cs [hunk 1]
```diff
@@ -138,7 +138,7 @@ public void AddAfterSelf(object? content)
         /// <exception cref="InvalidOperationException">
         /// Thrown if the parent is null.
         /// </exception>
-        public void AddAfterSelf(params object[] content)
+        public void AddAfterSelf(params object?[] content)
```
Comment (Jozkee) [inline src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNode.cs:141]: I think this triggers subtle nullability changes on the following snippet which I don't see anywhere in this PR.
https://github.com/dotnet/runtime/blob/1cc8b1c83ff6508dd8b428567e2160a8fd5a5081/src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XLinq.cs#L98-L109

**Review by jozkee — APPROVED** [review]
LGTM

File: src/libraries/System.Xml.XDocument/ref/System.Xml.XDocument.cs [hunk 2]
```diff
@@ -489,10 +489,10 @@ public partial class XStreamingElement
     {
         public XStreamingElement(System.Xml.Linq.XName name) { }
         public XStreamingElement(System.Xml.Linq.XName name, object content) { }
```
Comment (stephentoub) [inline src/libraries/System.Xml.XDocument/ref/System.Xml.XDocument.cs:491]: Should this be `object? content`?

**Comment by jeffhandley** [comment]
> cc: @jeffhandley for bar-check

The bar has raised notably since last week and we're now in recall-class bugs only. This fix will need to target 6.0 only, and I'll add it to dotnet/docs#21202, which aggregates the changes to annotations we're making in 6.0.

**Comment by jeffhandley** [comment]
I've added these changes to dotnet/docs#21202, including the latest suggestion from @stephentoub on `XStreamingElement(XName, Object)`. @krwq, once this is merged, please double-check the list posted in that issue and remove the `needs-breaking-change-doc-created` label once it's matched up.

File: src/libraries/System.Xml.XDocument/ref/System.Xml.XDocument.cs (hunk 2 above)
Comment (krwq) [reply src/libraries/System.Xml.XDocument/ref/System.Xml.XDocument.cs:491]: good catch, seems it's already `object?` elsewhere

**Comment by krwq** [comment]
@jeffhandley list in the issue seems correct to me

**Pull request merged**